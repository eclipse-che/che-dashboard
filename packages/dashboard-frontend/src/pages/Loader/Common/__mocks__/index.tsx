/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props } from '..';
import { LoaderTab } from '../../../../services/helpers/types';

export class CommonLoaderPage extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { alertItem, currentStepId, steps, activeTabKey, onTabChange } = this.props;
    const wizardSteps = steps.map(step => (
      <div key={step.id} data-testid={step.id}>
        <div data-testid="hasError">{step.hasError ? 'true' : 'false'}</div>
        <div data-testid="id">{step.id}</div>
        <div data-testid="title">{step.title}</div>
      </div>
    ));
    const actions = alertItem?.actionCallbacks?.map(actionCallback => (
      <div key={actionCallback.title} data-testid="action-callback">
        <button onClick={() => actionCallback.callback()}>{actionCallback.title}</button>
      </div>
    ));
    return (
      <div data-testid="common-loader-page">
        <button data-testid="tab-progress" onClick={() => onTabChange(LoaderTab.Progress)}>
          Progress
        </button>
        <button data-testid="tab-logs" onClick={() => onTabChange(LoaderTab.Logs)}>
          Logs
        </button>
        <div data-testid="active-tab-key">{activeTabKey}</div>
        <div data-testid="current-step-id">{currentStepId}</div>
        <div data-testid="alert">
          <span data-testid="alert-title">{alertItem?.title}</span>
          <span data-testid="alert-body">{alertItem?.children}</span>
        </div>
        <div data-testid="action-links">{actions}</div>
        <div data-testid="wizard-steps">{wizardSteps}</div>
      </div>
    );
  }
}
