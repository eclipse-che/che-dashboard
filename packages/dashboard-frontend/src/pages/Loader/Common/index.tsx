/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { PageSection, PageSectionVariants, Tab, Tabs } from '@patternfly/react-core';
import Head from '../../../components/Head';
import Header from '../../../components/Header';
import { LoaderStep } from '../../../components/Loader/Step';
import { AlertItem, DevWorkspaceStatus, LoaderTab } from '../../../services/helpers/types';
import { LoaderProgress } from '../../../components/Loader/Progress';
import { LoaderAlert } from '../../../components/Loader/Alert';
import WorkspaceLogs from '../../../components/WorkspaceLogs';
import { Workspace } from '../../../services/workspace-adapter';

import styles from './index.module.css';

export type Props = {
  activeTabKey: LoaderTab;
  alertItem: AlertItem | undefined;
  currentStepId: number;
  steps: LoaderStep[];
  workspace: Workspace | undefined;
  onTabChange: (tab: LoaderTab) => void;
};

export class CommonLoaderPage extends React.PureComponent<Props> {
  private handleTabClick(tabIndex: React.ReactText): void {
    this.props.onTabChange(tabIndex as LoaderTab);
  }

  render(): React.ReactNode {
    const { activeTabKey, alertItem, currentStepId, steps, workspace } = this.props;

    const pageTitle = workspace ? `Starting workspace ${workspace.name}` : 'Creating a workspace';
    const workspaceStatus = workspace?.status || DevWorkspaceStatus.STOPPED;
    const isToastAlert = activeTabKey === LoaderTab.Logs;
    const wizardSteps = LoaderStep.toWizardSteps(currentStepId, steps);

    return (
      <React.Fragment>
        <Head pageName={pageTitle} />
        <Header title={pageTitle} status={workspaceStatus} />
        <PageSection
          variant={PageSectionVariants.light}
          isFilled={true}
          className={styles.loaderPage}
        >
          <Tabs
            activeKey={activeTabKey}
            onSelect={(_event, tabIndex) => this.handleTabClick(tabIndex)}
            inset={{ default: 'insetLg' }}
            data-testid="loader-tabs"
          >
            <Tab
              eventKey={LoaderTab.Progress}
              title={LoaderTab[LoaderTab.Progress]}
              data-testid="loader-progress-tab"
              id="loader-progress-tab"
            >
              <PageSection>
                <LoaderAlert isToast={isToastAlert} alertItem={alertItem} />
                <LoaderProgress steps={wizardSteps} currentStepId={currentStepId} />
              </PageSection>
            </Tab>
            <Tab
              eventKey={LoaderTab.Logs}
              title={LoaderTab[LoaderTab.Logs]}
              data-testid="loader-logs-tab"
              id="loader-logs-tab"
            >
              <WorkspaceLogs
                workspaceUID={workspace?.uid}
                isDevWorkspace={workspace?.isDevWorkspace}
              />
            </Tab>
          </Tabs>
        </PageSection>
      </React.Fragment>
    );
  }
}
