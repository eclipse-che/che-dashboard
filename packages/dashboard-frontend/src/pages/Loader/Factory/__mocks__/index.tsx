/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export class FactoryLoaderPage extends React.PureComponent<Props, State> {
  render(): React.ReactNode {
    const { alertItem, currentStepId, steps } = this.props;
    const wizardSteps = steps.map(step => (
      <div key={step.id} data-testid={step.id}>
        <div data-testid="hasError">{step.hasError ? 'true' : 'false'}</div>
        <div data-testid="id">{step.id}</div>
        <div data-testid="title">{step.title}</div>
      </div>
    ));
    const alertActionLinks = alertItem?.actionCallbacks?.map(action => (
      <button onClick={() => action.callback()} key={action.title}>
        {action.title}
      </button>
    ));
    return (
      <div data-testid="factory-loader-page">
        <div data-testid="current-step-id">{currentStepId}</div>
        <div data-testid="alert">
          <span data-testid="alert-title">{alertItem?.title}</span>
          <span data-testid="alert-body">{alertItem?.children}</span>
          <div>{alertActionLinks}</div>
        </div>
        <div>{wizardSteps}</div>
      </div>
    );
  }
}
