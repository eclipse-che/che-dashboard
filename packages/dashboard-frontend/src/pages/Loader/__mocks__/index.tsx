/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { LoaderTab } from '@/services/helpers/types';

import { Props } from '..';

export class LoaderPage extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { workspace, tabParam, onTabChange } = this.props;
    return (
      <div data-testid="loader-page">
        <div data-testid="loader-tab">{tabParam}</div>
        <button data-testid="tab-button" onClick={() => onTabChange(LoaderTab.Events)} />
        <div data-testid="workspace">{workspace?.name || 'unknown'}</div>
      </div>
    );
  }
}
