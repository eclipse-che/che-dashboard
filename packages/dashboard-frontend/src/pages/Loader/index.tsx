/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { PageSection, PageSectionVariants, Tab, Tabs } from '@patternfly/react-core';
import React from 'react';
import Head from '../../components/Head';
import Header from '../../components/Header';
import { LoaderAlert } from '../../components/Loader/Alert';
import { LoaderProgress } from '../../components/Loader/Progress';
import { LoaderStep } from '../../components/Loader/Step';
import WorkspaceEvents from '../../components/WorkspaceEvents';
import WorkspaceLogs from '../../components/WorkspaceLogs';
import { AlertItem, DevWorkspaceStatus, LoaderTab } from '../../services/helpers/types';
import { Workspace } from '../../services/workspace-adapter';

import styles from './index.module.css';

export type Props = {
  alertItem: AlertItem | undefined;
  currentStepId: number;
  steps: LoaderStep[];
  tabParam: string | undefined;
  workspace: Workspace | undefined;
  onTabChange: (tabName: string) => void;
};

export type State = {
  activeTabKey: LoaderTab;
};

export class LoaderPage extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    const { tabParam } = this.props;
    const activeTabKey = tabParam && LoaderTab[tabParam] ? LoaderTab[tabParam] : LoaderTab.Progress;

    this.state = {
      activeTabKey,
    };
  }

  private handleTabClick(tabIndex: React.ReactText): void {
    const tabKey = tabIndex as LoaderTab;

    this.setState({
      activeTabKey: tabKey,
    });
    const tabName = LoaderTab[tabKey];
    this.props.onTabChange(tabName);
  }

  render(): React.ReactNode {
    const { alertItem, currentStepId, steps, workspace } = this.props;
    const { activeTabKey } = this.state;

    const pageTitle = workspace ? `Starting workspace ${workspace.name}` : 'Creating a workspace';
    const workspaceStatus = workspace?.status || DevWorkspaceStatus.STOPPED;
    const isToastAlert = activeTabKey === LoaderTab.Logs;
    const wizardSteps = LoaderStep.toWizardSteps(currentStepId, steps);

    return (
      <React.Fragment>
        <Head pageName={pageTitle} />
        <Header title={pageTitle} status={workspaceStatus} />
        <PageSection
          variant={PageSectionVariants.light}
          isFilled={true}
          className={styles.loaderPage}
        >
          <Tabs
            activeKey={activeTabKey}
            onSelect={(_event, tabIndex) => this.handleTabClick(tabIndex)}
            inset={{ default: 'insetLg' }}
            data-testid="loader-tabs"
          >
            <Tab
              eventKey={LoaderTab.Progress}
              title={LoaderTab[LoaderTab.Progress]}
              data-testid="loader-progress-tab"
              id="loader-progress-tab"
            >
              <PageSection>
                <LoaderAlert isToast={isToastAlert} alertItem={alertItem} />
                <LoaderProgress steps={wizardSteps} currentStepId={currentStepId} />
              </PageSection>
            </Tab>
            <Tab
              eventKey={LoaderTab.Logs}
              title={LoaderTab[LoaderTab.Logs]}
              data-testid="loader-logs-tab"
              id="loader-logs-tab"
            >
              <WorkspaceLogs workspaceUID={workspace?.uid} />
            </Tab>
            <Tab
              eventKey={LoaderTab.Events}
              title={LoaderTab[LoaderTab.Events]}
              data-testid="loader-events-tab"
              id="loader-events-tab"
            >
              <WorkspaceEvents workspaceUID={workspace?.uid} />
            </Tab>
          </Tabs>
        </PageSection>
      </React.Fragment>
    );
  }
}
