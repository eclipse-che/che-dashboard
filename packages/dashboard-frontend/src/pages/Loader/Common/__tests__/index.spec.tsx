/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { AlertVariant } from '@patternfly/react-core';
import { CommonLoaderPage } from '..';
import { LoadingStep, List, LoaderStep } from '../../../../components/Loader/Step';
import {
  getWorkspaceLoadingSteps,
  buildLoaderSteps,
} from '../../../../components/Loader/Step/buildSteps';
import { FakeStoreBuilder } from '../../../../store/__mocks__/storeBuilder';
import { DevWorkspaceBuilder } from '../../../../store/__mocks__/devWorkspaceBuilder';
import {
  AlertItem,
  LoaderTab,
  DevWorkspaceStatus,
  ActionCallback,
} from '../../../../services/helpers/types';
import devfileApi from '../../../../services/devfileApi';
import { constructWorkspace, Workspace } from '../../../../services/workspace-adapter';
import getComponentRenderer from '../../../../services/__mocks__/getComponentRenderer';

jest.mock('react-tooltip', () => {
  return function DummyTooltip(): React.ReactElement {
    return <div>Dummy Tooltip</div>;
  };
});
jest.mock('../../../../components/Loader/Alert');
jest.mock('../../../../components/Loader/Progress');
jest.mock('../../../../components/WorkspaceLogs');

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnTabChange = jest.fn();
const mockOnWorkspaceRestart = jest.fn();
const actionCallbacks: ActionCallback[] = [
  {
    title: 'Restart',
    callback: mockOnWorkspaceRestart,
  },
];

const namespace = 'user-che';
const workspaceName = 'wksp-test';
const currentStepId = LoadingStep.INITIALIZE;
const status: keyof typeof DevWorkspaceStatus = 'STARTING';
const activeTab = LoaderTab.Progress;

describe('Loader page', () => {
  let devWorkspace: devfileApi.DevWorkspace;
  let workspace: Workspace;
  let steps: List<LoaderStep>;
  let store: Store;

  beforeEach(() => {
    const loadingSteps = getWorkspaceLoadingSteps();
    steps = buildLoaderSteps(loadingSteps);
    devWorkspace = new DevWorkspaceBuilder()
      .withNamespace(namespace)
      .withName(workspaceName)
      .withStatus({ phase: status })
      .build();
    store = new FakeStoreBuilder()
      .withDevWorkspaces({
        workspaces: [devWorkspace],
      })
      .build();
    workspace = constructWorkspace(devWorkspace);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('component snapshot: creating a workspace', () => {
    const emptyStore = new FakeStoreBuilder().build();
    const snapshot = createSnapshot(emptyStore, {
      activeTab,
      steps: steps.values,
      workspace: undefined,
    });
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('component snapshot: starting a workspace', () => {
    const snapshot = createSnapshot(store, {
      activeTab,
      steps: steps.values,
      workspace,
    });
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should handle tab click', () => {
    renderComponent(store, {
      activeTab,
      steps: steps.values,
      workspace,
    });

    const tabButtonLogs = screen.getByRole('button', { name: 'Logs' });
    userEvent.click(tabButtonLogs);

    expect(mockOnTabChange).toHaveBeenCalledWith(LoaderTab.Logs);
  });

  it('should handle reload', () => {
    const alertItem: AlertItem = {
      key: 'alert-id',
      title: 'Failed to start the workspace',
      variant: AlertVariant.danger,
      actionCallbacks,
    };
    renderComponent(store, {
      activeTab,
      steps: steps.values,
      alertItem,
      workspace,
    });

    const reloadButton = screen.getByRole('button', { name: 'Restart' });
    userEvent.click(reloadButton);

    expect(mockOnWorkspaceRestart).toHaveBeenCalledWith();
  });

  it('should render Progress tab active by default', () => {
    renderComponent(store, {
      activeTab,
      steps: steps.values,
      workspace,
    });

    const tabpanelProgress = screen.queryByRole('tabpanel', { name: 'Progress' });
    const tabpanelLogs = screen.queryByRole('tabpanel', { name: 'Logs' });

    // active tab by default
    expect(tabpanelProgress).not.toBeNull();
    // disabled tab
    expect(tabpanelLogs).toBeNull();
  });

  it('should render Logs tab active', () => {
    renderComponent(store, {
      activeTab: LoaderTab.Logs,
      steps: steps.values,
      workspace,
    });

    const tabpanelProgress = screen.queryByRole('tabpanel', { name: 'Progress' });
    const tabpanelLogs = screen.queryByRole('tabpanel', { name: 'Logs' });

    // disabled tab
    expect(tabpanelProgress).toBeNull();
    // active tab
    expect(tabpanelLogs).not.toBeNull();
  });
});

function getComponent(
  store: Store,
  props: {
    steps: LoaderStep[];
    activeTab: LoaderTab;
    alertItem?: AlertItem;
    workspace?: Workspace;
  },
): React.ReactElement {
  return (
    <Provider store={store}>
      <CommonLoaderPage
        alertItem={props.alertItem}
        activeTabKey={props.activeTab}
        currentStepId={currentStepId}
        steps={props.steps}
        workspace={props.workspace}
        onTabChange={mockOnTabChange}
      />
    </Provider>
  );
}
