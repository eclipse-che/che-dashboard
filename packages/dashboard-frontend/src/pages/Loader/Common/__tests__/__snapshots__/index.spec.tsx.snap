// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader page component snapshot: creating a workspace 1`] = `
Array [
  <section
    className="pf-c-page__main-section pf-m-light"
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <div
        className="pf-l-stack__item"
      >
        <nav
          aria-label="Breadcrumb"
          className="pf-c-breadcrumb breadcrumb"
          data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
          data-ouia-component-type="PF4/Breadcrumb"
          data-ouia-safe={true}
        >
          <ol
            className="pf-c-breadcrumb__list"
          >
            <li
              className="pf-c-breadcrumb__item"
              onClick={[Function]}
            >
              <a
                className="pf-c-breadcrumb__link"
                href="/dashboard/"
                target={null}
              >
                Workspaces
              </a>
            </li>
            <li
              className="pf-c-breadcrumb__item"
            >
              <span
                className="pf-c-breadcrumb__item-divider"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </span>
              Creating a workspace
            </li>
          </ol>
        </nav>
      </div>
      <div
        className="pf-l-stack__item"
      >
        <div
          className="pf-l-flex"
        >
          <div
            className="pf-m-align-self-center"
          >
            <div
              className="pf-c-content"
            >
              <h1
                className=""
                data-pf-content={true}
              >
                Creating a workspace
              </h1>
            </div>
          </div>
          <div
            className=""
          >
            <span
              className="pf-c-label statusLabel stoppedLabel"
              data-tip=""
            >
              <span
                className="pf-c-label__content"
              >
                <span
                  className="pf-c-label__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill=""
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M8.001,0 C3.589,0 0,3.590 0,8 C0,12.410 3.590,16 8.001,16 C12.412,16 16,12.410
              16,8 C16,3.589 12.4125,0 8.001,0 Z M8,14 C4.690,14 2,11.310 2,8 C2,4.692 4.690,2
              8,2 C11.307,2 14,4.690 14,8 C14,11.309 11.307,14 8,14 Z"
                    />
                    <path
                      d="M8.001,3.5 C5.519,3.5 3.5,5.520 3.5,8 C3.5,10.481 5.520,12.5 8.000,12.5 C10.482,12.5 12.5,10.481
              12.5,8 C12.5,5.519 10.482,3.5 8.000,3.5 Z M8,10.75 C6.483,10.75 5.25,9.517 5.25,8 C5.25,6.484
              6.483,5.25 8,5.25 C9.516,5.25 10.75,6.483 10.75,8 C10.75,9.516 9.516,10.75 8,10.75 Z"
                    />
                  </svg>
                </span>
                Stopped
              </span>
            </span>
            <div>
              Dummy Tooltip
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="pf-c-page__main-section pf-m-light pf-m-fill loaderPage"
  >
    <div
      className="pf-c-tabs pf-m-inset-lg"
      data-ouia-component-id="OUIA-Generated-Tabs-1"
      data-ouia-component-type="PF4/Tabs"
      data-ouia-safe={true}
      data-testid="loader-tabs"
      onSelect={[Function]}
    >
      <button
        aria-hidden={false}
        aria-label="Scroll left"
        className="pf-c-tabs__scroll-button"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          />
        </svg>
      </button>
      <ul
        className="pf-c-tabs__list"
        onScroll={[Function]}
      >
        <li
          className="pf-c-tabs__item pf-m-current"
        >
          <button
            aria-controls="pf-tab-section-0-loader-progress-tab"
            className="pf-c-tabs__link"
            data-ouia-component-type="PF4/TabButton"
            data-ouia-safe={true}
            data-testid="loader-progress-tab"
            id="pf-tab-0-loader-progress-tab"
            onClick={[Function]}
          >
            Progress
          </button>
        </li>
        <li
          className="pf-c-tabs__item"
        >
          <button
            aria-controls="pf-tab-section-1-loader-logs-tab"
            className="pf-c-tabs__link"
            data-ouia-component-type="PF4/TabButton"
            data-ouia-safe={true}
            data-testid="loader-logs-tab"
            id="pf-tab-1-loader-logs-tab"
            onClick={[Function]}
          >
            Logs
          </button>
        </li>
      </ul>
      <button
        aria-hidden={false}
        aria-label="Scroll right"
        className="pf-c-tabs__scroll-button"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          />
        </svg>
      </button>
    </div>
    <section
      aria-labelledby="pf-tab-0-loader-progress-tab"
      className="pf-c-tab-content"
      data-ouia-component-type="PF4/TabContent"
      data-ouia-safe={true}
      hidden={false}
      id="pf-tab-section-0-loader-progress-tab"
      role="tabpanel"
      tabIndex={0}
    >
      <section
        className="pf-c-page__main-section"
      >
        <div
          data-testid="ide-loader-progress"
        >
          Dummy loading progress
        </div>
      </section>
    </section>
    <section
      aria-labelledby="pf-tab-1-loader-logs-tab"
      className="pf-c-tab-content"
      data-ouia-component-type="PF4/TabContent"
      data-ouia-safe={true}
      hidden={true}
      id="pf-tab-section-1-loader-logs-tab"
      role="tabpanel"
      tabIndex={0}
    >
      <div>
        Dummy Logs
      </div>
    </section>
  </section>,
]
`;

exports[`Loader page component snapshot: starting a workspace 1`] = `
Array [
  <section
    className="pf-c-page__main-section pf-m-light"
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <div
        className="pf-l-stack__item"
      >
        <nav
          aria-label="Breadcrumb"
          className="pf-c-breadcrumb breadcrumb"
          data-ouia-component-id="OUIA-Generated-Breadcrumb-2"
          data-ouia-component-type="PF4/Breadcrumb"
          data-ouia-safe={true}
        >
          <ol
            className="pf-c-breadcrumb__list"
          >
            <li
              className="pf-c-breadcrumb__item"
              onClick={[Function]}
            >
              <a
                className="pf-c-breadcrumb__link"
                href="/dashboard/"
                target={null}
              >
                Workspaces
              </a>
            </li>
            <li
              className="pf-c-breadcrumb__item"
            >
              <span
                className="pf-c-breadcrumb__item-divider"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </span>
              Starting workspace wksp-test
            </li>
          </ol>
        </nav>
      </div>
      <div
        className="pf-l-stack__item"
      >
        <div
          className="pf-l-flex"
        >
          <div
            className="pf-m-align-self-center"
          >
            <div
              className="pf-c-content"
            >
              <h1
                className=""
                data-pf-content={true}
              >
                Starting workspace wksp-test
              </h1>
            </div>
          </div>
          <div
            className=""
          >
            <span
              className="pf-c-label pf-m-blue statusLabel"
              data-tip=""
            >
              <span
                className="pf-c-label__content"
              >
                <span
                  className="pf-c-label__icon"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="rotate"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M513.417211,16.013668 L513.417211,112.013668 C513.406007,120.539653 520.082422,127.576718 528.597211,128.013668 C732.697211,136.513668 896.147211,305.013668 896.147211,511.343668 C896.147211,723.013668 724.007211,895.163668 512.417211,895.163668 C437.186239,895.277345 363.602574,873.135095 300.927211,831.523668 C294.58293,827.2982 286.138663,828.135095 280.747211,833.523668 L211.807211,902.353668 C208.502935,905.676185 206.82123,910.280199 207.20607,914.950237 C207.590911,919.620275 210.00361,923.886884 213.807211,926.623668 C300.709573,989.398037 405.213535,1023.13146 512.417211,1023.01398 C794.537211,1023.01398 1023.91724,793.433668 1023.91724,511.413668 C1024.00721,235.103668 804.007211,9.22366802 529.897211,0.00645736761 C525.571205,-0.116171039 521.377128,1.51204372 518.271915,4.52681369 C515.166702,7.54158367 513.415263,11.6857144 513.417211,16.013668 M97.1272107,212.923668 C64.2190689,258.479443 39.0698895,309.162997 22.6972107,362.923668 C21.2705092,367.756559 22.2039184,372.979173 25.2163356,377.018685 C28.2287528,381.058197 32.9681443,383.442545 38.0072107,383.453683 L139.407211,383.453683 C145.970195,383.46249 151.873012,379.462528 154.297211,373.363668 C164.307669,347.593304 177.068112,322.977652 192.357211,299.943668 C196.579666,293.633658 195.741339,285.216847 190.357211,279.863668 L121.497211,211.013668 C118.186635,207.662531 113.567544,205.940538 108.871166,206.30669 C104.174788,206.672841 99.8783994,209.089927 97.1272107,212.913668 M112.907211,511.433663 L17.0072107,511.433663 C12.679257,511.433663 8.5351263,513.183159 5.52035633,516.288373 C2.50558635,519.393586 0.877371594,523.587662 1.00721065,527.913668 C4.13104542,629.408217 37.6085847,727.635936 97.1172107,809.913668 C99.8539943,813.717268 104.120603,816.129968 108.790642,816.514809 C113.46068,816.899649 118.064693,815.217944 121.387211,811.913668 L190.217211,743.073668 C195.605784,737.682216 196.442679,729.237949 192.217211,722.893668 C153.520825,664.580171 131.611534,596.745437 128.887211,526.813668 C128.553964,518.220138 121.487197,511.427209 112.887211,511.433663"
                    />
                  </svg>
                </span>
                Starting
              </span>
            </span>
            <div>
              Dummy Tooltip
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="pf-c-page__main-section pf-m-light pf-m-fill loaderPage"
  >
    <div
      className="pf-c-tabs pf-m-inset-lg"
      data-ouia-component-id="OUIA-Generated-Tabs-2"
      data-ouia-component-type="PF4/Tabs"
      data-ouia-safe={true}
      data-testid="loader-tabs"
      onSelect={[Function]}
    >
      <button
        aria-hidden={false}
        aria-label="Scroll left"
        className="pf-c-tabs__scroll-button"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          />
        </svg>
      </button>
      <ul
        className="pf-c-tabs__list"
        onScroll={[Function]}
      >
        <li
          className="pf-c-tabs__item pf-m-current"
        >
          <button
            aria-controls="pf-tab-section-0-loader-progress-tab"
            className="pf-c-tabs__link"
            data-ouia-component-type="PF4/TabButton"
            data-ouia-safe={true}
            data-testid="loader-progress-tab"
            id="pf-tab-0-loader-progress-tab"
            onClick={[Function]}
          >
            Progress
          </button>
        </li>
        <li
          className="pf-c-tabs__item"
        >
          <button
            aria-controls="pf-tab-section-1-loader-logs-tab"
            className="pf-c-tabs__link"
            data-ouia-component-type="PF4/TabButton"
            data-ouia-safe={true}
            data-testid="loader-logs-tab"
            id="pf-tab-1-loader-logs-tab"
            onClick={[Function]}
          >
            Logs
          </button>
        </li>
      </ul>
      <button
        aria-hidden={false}
        aria-label="Scroll right"
        className="pf-c-tabs__scroll-button"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          />
        </svg>
      </button>
    </div>
    <section
      aria-labelledby="pf-tab-0-loader-progress-tab"
      className="pf-c-tab-content"
      data-ouia-component-type="PF4/TabContent"
      data-ouia-safe={true}
      hidden={false}
      id="pf-tab-section-0-loader-progress-tab"
      role="tabpanel"
      tabIndex={0}
    >
      <section
        className="pf-c-page__main-section"
      >
        <div
          data-testid="ide-loader-progress"
        >
          Dummy loading progress
        </div>
      </section>
    </section>
    <section
      aria-labelledby="pf-tab-1-loader-logs-tab"
      className="pf-c-tab-content"
      data-ouia-component-type="PF4/TabContent"
      data-ouia-safe={true}
      hidden={true}
      id="pf-tab-section-1-loader-logs-tab"
      role="tabpanel"
      tabIndex={0}
    >
      <div>
        Dummy Logs
      </div>
    </section>
  </section>,
]
`;
